File: frontend/scripts/env-check.js
#!/usr/bin/env node
const required = [
  'VITE_APP_NAME',
  'VITE_FRONTEND_URL',
  'VITE_DEFAULT_LOCALE',
  'VITE_SUPPORTED_LOCALES',
  'VITE_API_URL',
  'VITE_SUPABASE_URL',
  'VITE_SUPABASE_ANON_KEY'
]

const missing = required.filter(k => !process.env[k])
if (missing.length) {
  console.error('Missing required environment variables:', missing.join(', '))
  process.exit(1)
} else {
  console.log('All required VITE_ env variables are set.')
}

File: backend/scripts/env_check.py
#!/usr/bin/env python3
import os
import sys

REQUIRED_KEYS = [
    'SUPABASE_URL',
    'SUPABASE_ANON_KEY',
    'OPENAI_API_KEY',
    'STRIPE_SECRET_KEY',
    'STRIPE_WEBHOOK_SECRET',
    'BACKEND_URL',
    'FRONTEND_URL',
    'JWT_SECRET_KEY',
]

missing = [k for k in REQUIRED_KEYS if not os.getenv(k)]
if missing:
    print('Missing required environment variables:', ', '.join(missing))
    sys.exit(1)
else:
    print('All required environment variables are set.')
    sys.exit(0)
